### PR TITLE
Link the `test_files` to GitHub

### PR DESCRIPTION
**Pull request by vasudev-sharma**
## Checklist

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [ ] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [ ] I've added relevant tests for my contribution
- [ ] I've updated the documentation and/or added correct docstrings
- [x] I've assigned a reviewer
- [x] I've consulted [ADS's internal developer documentation](https://github.com/neuropoly/axondeepseg/wiki/Guidelines-for-contribution) to ensure my contribution is in line with any relevant design decisions

## Description

Similar to ADS models which were hosted on OSF, this PR takes care of the test files(`__test_files__`) which were hosted on OSF (https://osf.io/vqt94/?action=download) earlier and now being hosted on GitHub (https://github.com/axondeepseg/data-testing). This PR aims to download these test files hosted on GH to ADS.

## Testing
No additional tests are required. The previous unit test [test_download_test.py](https://github.com/neuropoly/axondeepseg/blob/master/test/test_download_tests.py) was sufficient for this change.

### CODE REVIEW

File: AxonDeepSeg/download_tests.py [hunk 1]
```diff
@@ -10,22 +10,27 @@ def download_tests(destination = None):
         destination = convert_path(destination)
         test_files_destination = destination / "__test_files__"
 
-    url_tests = "https://osf.io/vqt94/?action=download"  # URL of TEM model hosted on OSF storage with the appropriate versioning on OSF
+
+    # retrieve latest release identifier from github API
+    response_tests_files = requests.get("https://api.github.com/repos/axondeepseg/data-testing/releases/latest")
+    folder_name_test_files = Path("data-testing-"+str(response_tests_files.json()['tag_name']))
+
+    url_tests = "https://github.com/axondeepseg/data-testing/archive/refs/tags/"+str(response_tests_files.json()['tag_name'])+".zip"   # URL of TEM model hosted on OSF storage with the appropriate versioning on OSF
```
Comment (hermancollin) [inline AxonDeepSeg/download_tests.py:18]: If I am not mistaken, for the test files, as for #533, we will need the tag_name hardcoded in the URL.
If this is indeed the case I suggest you use the same trick I used (comparing files in the current directory before/after the download).

Comment (vasudev-sharma) [reply AxonDeepSeg/download_tests.py:18]: > If I am not mistaken, for the test files, as for #533, we will need the tag_name hardcoded in the URL.
If this is indeed the case I suggest you use the same trick I used (comparing files in the current directory before/after the download).

Yes, you're right. It needs to be updated. I'll update this. 

> edit: also if we indeed want a static tag_name in the URL, we should wait for @Stoyan-I-A to modify the files in the gh repo before merging this PR

Yes you are absolutely right, we should wait for @Stoyan-I-A or me to update the data-testing repo.

**Review by hermancollin — APPROVED** [review]
tested it and it works perfectly :clap:

**Comment by vasudev-sharma** [comment]
**NOTE:**⚠️Don't merge this PR right now.⚠️

Merge it when the comment below has been addressed.
> Yes you are absolutely right, we should wait for @Stoyan-I-A or me to upload additional images in the data-testing repo

**Comment by Stoyan-I-A** [comment]
Thank you @vasudev-sharma for this PR. I don't have the `push access` required to upload files into [axondeepseg/data-testing](https://github.com/axondeepseg/data-testing). I will ask for permission.

**Comment by Stoyan-I-A** [comment]
Somehow adding me as an editor didn't give me access to `push access`. Nevertheless, I created a PR to add the files I require: https://github.com/axondeepseg/data-testing/pull/1

**Comment by jcohenadad** [comment]
> Somehow adding me as an editor didn't give me access to `push access`. Nevertheless, I created a PR to add the files I require: [axondeepseg/data-testing#1](https://github.com/axondeepseg/data-testing/pull/1)

i'll re-invite you, for some reasons it didn't go through:
image

**Comment by Stoyan-I-A** [comment]
I received the invitation! I think we can merge this PR now.

**Comment by hermancollin** [comment]
@Stoyan-I-A I see you've merged your PR on the data-testing repo and merged this one as well. Only problem is that the testing data hasn't been re-released on github, so ADS will still download the old release (without your 2 new testing images).

**Comment by vasudev-sharma** [comment]
> @Stoyan-I-A I see you've merged your PR on the data-testing repo and merged this one as well. 🎉
> 
> Only problem is that the testing data hasn't been re-released on github, so ADS will still download the old release (without your 2 new testing images).

@hermancollin would like to create a new release of data-testing? I'll open a quick PR to fix this.

**Comment by hermancollin** [comment]
@vasudev-sharma I already created a new release located here
https://github.com/axondeepseg/data-testing/releases/tag/r20210625

**Comment by vasudev-sharma** [comment]
> My suggestion is we could update this tag name in #519 (maybe it's more logical to update the tag name directly in the concerned PR?)

It is better a create a quick PR to resolve this because #519 is completely separate PR. Created PR #535 for the same.

**Pull request merged**